components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -294,12 +294,27 @@ OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumerAppDTO
             if (StringUtils.isNotEmpty(tenantAwareLoggedInUsername)) {
                 defaultAppOwner = buildAuthenticatedUser(tenantAwareLoggedInUsername, tenantDomain);
             } else {
-                Optional<User> tenantAwareLoggedInUser = OAuthUtil.getUser(tenantDomain, null);
+                Optional<User> tenantAwareLoggedInUser = OAuthUtil.getUser(tenantDomain, application.getUsername());
                 if (tenantAwareLoggedInUser.isPresent()) {
                     defaultAppOwner = new AuthenticatedUser(tenantAwareLoggedInUser.get());
                 }
             }
 
+            /*
+             * If there is no authenticated user, it is due to the DCR endpoint api authentication being turned off and
+             * hence we are setting the tenant admin as authenticated the app owner.
+             * If DCR endpoint api authentication is enabled there should be an authenticated user at this point,
+             * since if not, there will be an error from Authentication valve above this level.
+             */
+            if (defaultAppOwner == null) {
```

Comment:
evaluate for privilege escalations

Reply:
checked the following flows where new apps are created.

- new app creation via UI
- new org creation via UI

There is already a `defaultAppOwner` for each above. So no privilege escalations from those scenarios. The only place I found there wont be an app owner is when creating an app through DCR after turning off authentication for DCR. Currently it gives a server error in that scenario since `defaultAppOwner` is null. With this PR, it is fixed and the tenant admin is set for the `defaultAppOwner`.